estransport/estransport.go

```diff
@@ -100,11 +104,31 @@ type Client struct {
 //
 // http.DefaultTransport will be used if no transport is passed in the configuration.
 //
-func New(cfg Config) *Client {
+func New(cfg Config) (*Client, error) {
 	if cfg.Transport == nil {
 		cfg.Transport = http.DefaultTransport
 	}
 
+	if cfg.CACert != nil {
+		httpTransport, ok := cfg.Transport.(*http.Transport)
+		if !ok {
+			return nil, fmt.Errorf("unable to set CA certificate for transport of type %T", cfg.Transport)
+		}
+
+		sysCertPool, err := x509.SystemCertPool()
```

Comment:
Perhaps check with other clients, but I would strongly suggest using `x509.NewCertPool()` here, so the specified CA certs are the _only_ ones that are used for verification.

Reply:
Addressed in dcdb7f2.